### DISCUSSION TITLE
How to write a good question in Slack

### DISCUSSION THREAD

**Discussion by kyleaton** [great-expectations/great_expectations#4951] [General]
I just want to quickly highlight some of the components of an effective support question in the Great Expectations Community. An effective support question makes it easy for someone to understand exactly what the problem is so they can quickly provide a helpful answer that everyone can learn from. An effective question is also formatted appropriately for the community to help us stay organized and efficient. You don’t need to do everything below for every question, but it will help the community out if people at least go through this checklist mentally as they are thinking about posting support questions.

An effective question quickly conveys the relevant information to understand the problem.

1. What’s your environment? (e.g. OS, GE Version, stack, relevant configurations, etc.)
2. What’s the context within which you’re asking the question? (e.g. what are you trying to do, what guides have you followed, what have you searched for in the documentation, slack, and Discuss to solve the issue on your own, what have you tried so far, what have you thought about trying but haven’t yet tried, etc.)
3. What is the actual problem? (e.g. what commands or code did you run that led to the issue, what did you expect, what are you actually observing, stack trace, etc.)
4. Is there anything else someone would need to know to reproduce the issue?

An effective question uses community guidelines for formatting:

Succinct - As long as it needs to be, but as short as possible.

1. Threads - Please use threads for questions in slack. Publishing multiple channel-level posts for the same question just makes it harder to keep track of what needs to be answered.
2. Ask things once - posting the same question in multiple places just makes more work for the community and the Developer Relations team. Adding a DM about the same question again just adds further work.
3. Ask support questions in public - The community benefits from the questions and answers that everyone posts. Please keep good questions public. DMing is useful for coordinating and providing feedback very specific to your organization. For the most part, support questions should be in public.
4. [Read the guidelines](https://github.com/great-expectations/great_expectations/blob/develop/SLACK_GUIDELINES.md) to see what type of content goes in which channels. It helps to follow those guidelines.
5. If you have other suggestions, please post them in the comments here!

**Comment by a-chumagin** [comment]
my 5 cents
add details to the question: 
- Version of GE
- Error message - not screenshot, please. Text is better
- Data set example
- Code snipped (we should understand it is for instance Pandas or Spark)
